File: twidere/src/androidTest/java/org/mariotaku/twidere/model/UserKeyTest.java
package org.mariotaku.twidere.model;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Created by mariotaku on 16/3/9.
 */
public class UserKeyTest {

    @Test
    public void testToString() throws Exception {
        assertEquals("abc@twitter.com", new UserKey("abc", "twitter.com").toString());
    }

    @Test
    public void testValueOf() throws Exception {
        assertEquals(UserKey.valueOf("abc@twitter.com"), new UserKey("abc", "twitter.com"));
    }
}

File: twidere/src/androidTest/java/org/mariotaku/twidere/model/UserKeyTest.kt
package org.mariotaku.twidere.model

import org.junit.Test

import org.junit.Assert.assertEquals

/**
 * Created by mariotaku on 16/3/9.
 */
class UserKeyTest {

    @Test
    @Throws(Exception::class)
    fun testToString() {
        assertEquals("abc@twitter.com", UserKey("abc", "twitter.com").toString())
        assertEquals("\\@user@twitter.com", UserKey("@user", "twitter.com").toString())
        assertEquals("\\@u\\\\ser@twitter.com", UserKey("@u\\ser", "twitter.com").toString())
    }

    @Test
    @Throws(Exception::class)
    fun testValueOf() {
        assertEquals(UserKey("abc", "twitter.com"), UserKey.valueOf("abc@twitter.com"))
        assertEquals(UserKey("abc@", "twitter.com"), UserKey.valueOf("abc\\@@twitter.com"))
        assertEquals(UserKey("abc@", "twitter.com"), UserKey.valueOf("a\\bc\\@@twitter.com"))
        assertEquals(UserKey("a\\bc@", "twitter.com"), UserKey.valueOf("a\\\\bc\\@@twitter.com"))
    }
}